apps/extension/src/background/keyring/keyring.ts

```diff
@@ -1,68 +1,217 @@
 import { KVStore } from "@anoma/storage";
+import { AEAD, Bip44, Mnemonic, PhraseSize } from "@anoma/crypto";
+import { Address } from "@anoma/shared";
+import { KeyRingStatus, MnemonicState, AccountState } from "./types";
+import { v5 as uuid } from "uuid";
+import { Bip44Path, DerivedAccount } from "./types";
 
-export enum KeyRingStatus {
-  NOTLOADED,
-  EMPTY,
-  LOCKED,
-  UNLOCKED,
+import { IStore, Store } from "../types";
+
+// Generated UUID namespace for uuid v5
+const UUID_NAMESPACE = "9bfceade-37fe-11ed-acc0-a3da3461b38c";
+
+// Construct unique uuid, passing in an arbitray number of arguments.
+// This could be a unique parameter of the object receiving the id,
+// or an index based on the number of existing objects in a hierarchy.
+const getId = (name: string, ...args: (number | string)[]): string => {
+  // Ensure a unique UUID
+  let uuidName = name;
+
+  // Concatenate any number of args onto the name parameter
+  args.forEach((arg) => {
+    uuidName = `${uuidName}::${arg}`;
+  });
+
+  return uuid(uuidName, UUID_NAMESPACE);
+};
+
+enum KeyRingStore {
```

Comment:
nit: would it be more correct to call these `<name>StoreKey` as they are not the store, but a key to a store. The method they are passed also names the parameter `key`.

```suggestion
enum KeyRingStoreKey {
```

Reply:
@memasdeligeorgakis This enum has been removed entirely. In https://github.com/anoma/namada-interface/pull/97 (storage updates), everything resides under a single store, so there's no need to differentiate. Instead, there's a single constant defining the store's key (see https://github.com/anoma/namada-interface/blob/1bcb3230792fb0d72f27e962f462b6272668783a/apps/extension/src/background/keyring/keyring.ts#L30-L40).